internal/cli/compile/compile.go

```diff
@@ -149,17 +145,14 @@ func runCompileCommand(cmd *cobra.Command, args []string) {
 	logrus.Info("Executing `arduino-cli compile`")
 
 	if dumpProfile && feedback.GetFormat() != feedback.Text {
-		feedback.Errorf(tr("You cannot use the %[1]s flag together with %[2]s.", "--dump-profile", "--format json"))
-		os.Exit(errorcodes.ErrBadArgument)
+		feedback.Fatal(tr("You cannot use the %[1]s flag together with %[2]s.", "--dump-profile", "--format json"), feedback.ErrBadArgument)
 	}
```

Comment:
Thanks to the fabulous advancements made in this PR, this restriction is no longer necessary. 

This check can be removed if the following lines are also removed:

https://github.com/arduino/arduino-cli/blob/4330894d23125ebff45afce017e8bf3962ddda2b/internal/cli/compile/compile.go#L311-L312

Reply:
Ok, finally I removed the check as you suggested, but I've also put the generated profile in his own field in the result... at least in the JSON output, it will be clearly separated from the rest.